manifests/exports.pp

```diff
@@ -0,0 +1,22 @@
+# Class nfs::exports
+# implements export resource
+# it should work even if empty but if you want to put exports here then do the following
+# Class{ nfs::exports:
+#           export1    => {
+#               $guest => "xyz.example.com",
+#               $share => "/nfs/share/",
+#               $options = "rootsquash,rw",
+#
+#            },
+#         export2 => {}
+#       }
+class nfs::exports($exporthash = {},
+                  $hieraarray = false)
+{
+  if $hieraarray == true {
+    create_resources(::nfs::export, hiera_array(::nfs::exports::exporthash, $exporthash))
```

Comment:
Right, this kind of lookup should go in the profile layer and be passed down to the component module as a parameter.